compiler/rustc_trait_selection/src/traits/project.rs

```diff
@@ -1856,15 +1874,23 @@ fn confirm_impl_candidate<'cx, 'tcx>(
     let substs =
         translate_substs(selcx.infcx(), param_env, impl_def_id, substs, assoc_ty.defining_node);
     let ty = tcx.type_of(assoc_ty.item.def_id);
+    let term: ty::Term<'tcx> = if INTO_CONST {
+        // FIXME(associated_const_equality): what are the right substs?
```

Comment:
you'll want identity substs, so `InternalSubsts::identity_for_item(tcx, assoc_ty.item.def_id)`